hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/RolloutHelper.java

```diff
@@ -215,30 +239,69 @@ static String getAllGroupsTargetFilter(final List<RolloutGroup> groups) {
         if (groups.stream().anyMatch(group -> StringUtils.isEmpty(group.getTargetFilterQuery()))) {
             return "";
         }
-        return groups.stream().map(RolloutGroup::getTargetFilterQuery).collect(Collectors.joining(","));
+
+        return "(" + groups.stream().map(RolloutGroup::getTargetFilterQuery).distinct().sorted()
```

Comment:
It probably does not matter in this case how it is used, because the groups list is always in the same order for multiple calls of the `getAllGroupsTargetFilter` method.
The result is later used for caching and thus in the key of a map, where I want't to uniquely identify the filter. I wanted to avoid two filters in the Map, that mean the same but are ordered differently. Maybe there is a corner case where the `.sorted()` call helps.